manager/src/main/java/io/gingersnapproject/metrics/CacheManagerMetrics.java

```diff
@@ -15,4 +17,11 @@ public interface CacheManagerMetrics {
     */
    <T> CacheAccessRecord<T> recordCacheAccess();
 
+   /**
+    * Register metrics associated with the {@code rule} stored in {@code cache}.
+    *
+    * @param rule  The rule's name.
+    * @param cache The {@link LoadingCache} where the data is cached.
+    */
+   void registerRulesMetric(String rule, LoadingCache<?, ?> cache);
```

Comment:
We also need the ability to deregister a Rule's metric for when they're removed at runtime. This will be called by the ConfigMap informer.

Reply:
is it implemented? I couldn't find anything yet.

Reply:
I think we need to look at everything once we have the informer. I think it is fine for now.

Reply:
A draft of the Informer PR is already up https://github.com/gingersnap-project/cache-manager/pull/25. This PR will land first, so if we add this additional method it should just be a case of @rigazilla calling it via his runtime configuration abstraction on a delete event.